torchao/quantization/autoquant.py

```diff
@@ -263,10 +269,47 @@ def _autoquant_test(cls, act_mat, weight, bias, best_time, mode=["relu", None]):
         print(f">>time: {res:0.3f}ms for {cls}, to_beat: {best_time:0.3f}ms ")
         return res
 
-class AQInt8DynamicallyQuantizedLinearWeight(AQMixin, Int8DynamicallyQuantizedLinearWeight):
+class AQInt8DynamicallyQuantizedLinearWeight(AQMixin, LinearActivationQuantizedTensor):
     """
     AutoQuantizable version of Int8DynamicallyQuantizedLinearWeight
     """
+    @classmethod
+    def from_float(cls, weight):
+        # in_features = weight.shape[1]
+        # int8 dynamic quantization only has benefit when in_feature > 16
+        # if in_features <= 16:
+            # return weight
+
+        # avoid circular dep
+        from torchao.dtypes import to_affine_quantized
```

Comment:
can we reuse the code from quant_api.py? also we need to refactor the `input_quant_func` to be a normal function (not lambda) in order for serialization to work I think, might help to do that refactor at the same time

Reply:
the issue is that we need to call from_float with super in order to get this to work correctly.  The code in quant_api would generate an aqt, not an autoquant class inheriting from an aqt. (I tried this approach initially since that's how it worked with subclass)

If we want to reuse the code, it may make sense to make a function that like prepares all the variables needed to go into from_float

def int8_weight_only_kwargs(weight):
       ...do the code up to to_affine_quantized and put it all together
       return a_bunch_of_kwargs
       
then you could have the quant_api code be like

def int8_weight_only():
    def apply_int8wo_quant(weight):
        kwargs = int8_weight_only_kwargs(weight)   
        return to_affine_quantized(**kwargs)
    return _get_linear_subclass_inserter(apply_int8wo_quant)

then in autoquant we could do similar

def from_float(weight):
      kwargs = int8_weight_only_kwargs(weight)
      super().from_float(**kwargs)

Reply:
we can move the apply quant to weight Tensor function to top level as well I think